src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreMetricsInstrumentationOptions.cs

```diff
@@ -14,6 +14,7 @@
 // limitations under the License.
 // </copyright>
 
+#if !NET8_0_OR_GREATER
```

Comment:
@vishweshbankwar I think this whole thing is a bad idea to do. We don't want to expose different APIs for different targets. That makes our NuGet toxic. Let's say someone codes against the `netstandard2.0` target in a shared library. These APIs will be there and compile. Then that shared library is consumed into a `net8.0` host. Suddenly that shared library won't compile. Best-practice is to expose the same API for all your targets. IMO we should just document Filter & Enrich don't fire on .NET 8.

Reply:
Regarding Enrich and Filter: See https://github.com/open-telemetry/opentelemetry-dotnet/issues/4932. I will do a separate PR for that.

> IMO we should just document Filter & Enrich don't fire on .NET 8.

I am ok to do this for unblocking this PR.

Reply:
This PR can't be merged because it is breaking the API from 7 to 8. The package validation tool is doing a nice job of catching that and blocking it 🧱 What you need to do is remove the `#if !NET8_OR_GREATER` added around anything `public`. We need the public APIs in 8 to match 7 (and others). You can do the documentation in a follow-up, that's fine by me 😄

Reply:
Why not start with removing enrich/filter first, and then land this PR?

Reply:
I plan to do that after this PR. have fixed the APIs for now.